ABDataCollectionCore.js

```diff
@@ -556,27 +556,35 @@ module.exports = class ABDataCollectionCore extends ABMLClass {
    /**
     * @method refreshLinkCursor
     * filter data in data view by match id of link data view
-    *
+    * @param {bool} shouldReloadData - boolean that is passed if new data should be fetched from server
     */
-   refreshLinkCursor() {
-      // if we are loading the data server side already filtered we do not need to fitler it client side
-      if (this.__reloadWheres && !this.settings.loadAll) return false;
-
-      let linkCursor;
-      let dvLink = this.datacollectionLink;
-      if (dvLink) {
-         linkCursor = dvLink.getCursor();
-      }
+   refreshLinkCursor(shouldReloadData = false) {
+      // if this data collection is being passed in new where conditions because of server side binding we need to reload the data
+      // and if this data colleciton is not loading all data from the server on init we need to reload data from server
+      // finally if we tell the data collection that we should reload the data because of an update we should force the reload
+      // the final check happens when a data collection doesn't have a new set cursor but one of its children has updated causing it to be stale (ex: calculated fields)
+      if (!this.settings.loadAll && shouldReloadData) {
+         console.log("reload data collection: ", this.id);
```

Comment:
let's not keep dumping console.log() statements.

Reply:
Good call...it was only helpful during testing...not needed in production.